diff -Naur upower-v1.90.0/src/linux/up-device-supply.c upower-v1.90.0-2/src/linux/up-device-supply.c
--- upower-v1.90.0/src/linux/up-device-supply.c	2024-02-13 04:47:45.985364890 +0100
+++ upower-v1.90.0-2/src/linux/up-device-supply.c	2024-02-17 18:47:36.453532879 +0100
@@ -58,13 +58,16 @@
 	GUdevDevice *native;
 	gboolean online_old, online_new;
 
+        g_warning ("DEBUG: UP_DEVICE_KIND_LINE_POWER updating");
+
 	/* get new AC value */
 	native = G_UDEV_DEVICE (up_device_get_native (device));
 
 	g_object_get (device,
 		      "online", &online_old,
 		      NULL);
-	online_new = g_udev_device_get_sysfs_attr_as_int_uncached (native, "online");
+        online_new = g_udev_device_get_sysfs_attr_as_int_uncached (native, "online");
+        g_warning ("DEBUG: UP_DEVICE_KIND_LINE_POWER state: %d", online_new);
 	/* Avoid notification if the value did not change. */
 	if (online_old != online_new)
 		g_object_set (device,
@@ -442,11 +445,13 @@
 			type = UP_DEVICE_KIND_LINE_POWER;
 		else if (g_strstr_len (native_path, -1, "wacom_") != NULL)
 			type = UP_DEVICE_KIND_TABLET;
-		else
+		else {
 			g_warning ("USB power supply %s without usb_type property, please report",
 				   native_path);
+                        type = UP_DEVICE_KIND_LINE_POWER;
+                }
 	} else {
-		g_warning ("did not recognise type %s, please report", device_type);
+		g_warning ("did not recognise type %s, please report Assuming UP_DEVICE_KIND_LINE_POWER", device_type);
 	}
 
 out:
@@ -537,6 +542,8 @@
 	g_object_get (device,
 		      "type", &type,
 		      NULL);
+        g_warning("Refresh Type: %d", type);
+
 	if (type == UP_DEVICE_KIND_LINE_POWER) {
 		updated = up_device_supply_refresh_line_power (supply, reason);
 	} else {
diff -Naur upower-v1.90.0/src/org.freedesktop.UPower.conf.in upower-v1.90.0-2/src/org.freedesktop.UPower.conf.in
--- upower-v1.90.0/src/org.freedesktop.UPower.conf.in	2024-02-13 04:47:45.997365114 +0100
+++ upower-v1.90.0-2/src/org.freedesktop.UPower.conf.in	2024-02-04 20:41:56.206243724 +0100
@@ -4,6 +4,10 @@
  "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
  "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
 <busconfig>
+  <policy user="%DISTRO%">
+    <allow own="org.freedesktop.UPower"/>
+  </policy>
+
   <!-- Only root can own the service -->
   <policy user="root">
     <allow own="org.freedesktop.UPower"/>
