diff -Naur jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef/nvmpi_dec.cpp jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef-2/nvmpi_dec.cpp
--- jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef/nvmpi_dec.cpp	2022-02-11 02:06:36.274569167 +0100
+++ jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef-2/nvmpi_dec.cpp	2022-02-11 02:07:19.427282669 +0100
@@ -498,7 +498,7 @@
 	ctx->dec->capture_plane.setStreamStatus(false);
 	
 	if (ctx->dec_capture_loop) {
-		ctx->dec_capture_loop->join();
+		//ctx->dec_capture_loop->join();
 		delete ctx->dec_capture_loop;
 		ctx->dec_capture_loop = nullptr;
 	}
