diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ar.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ar.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ar.h	2023-04-04 05:39:02.874217001 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ar.h	2023-04-04 05:31:23.447652065 +0200
@@ -9211,18 +9211,7 @@
    MENU_ENUM_LABEL_VALUE_INPUT_BLOCK_TIMEOUT,
    "مهلة كتلة الإدخال"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU فوق الساعة"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "تبديل GPU على مدار الساعة أو تحت الساعة."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "تجاوز الساعة المعالج"
@@ -9294,6 +9283,8 @@
    "ملف تكوين خاطئ - تعذر العثور على APNAME أو PASSWORD في %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ast.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ast.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ast.h	2023-04-04 05:39:02.878217093 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ast.h	2023-04-04 05:31:23.447652065 +0200
@@ -3739,10 +3739,7 @@
    MSG_READ_ONLY,
    "Estáu del almacenamientu internu: namás llectura"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -3778,6 +3775,8 @@
    "El ficheru de configuración ye incorreutu - nun se pudo atopar APNAME o PASSWORD en %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ca.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ca.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ca.h	2023-04-04 05:39:02.878217093 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ca.h	2023-04-04 05:31:23.447652065 +0200
@@ -8179,18 +8179,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Augmenta o redueix la brillantor de la pantalla."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Força la GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Augmenta o redueix la velocitat de rellotge de la GPU de la Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Força la CPU"
@@ -8274,6 +8263,8 @@
    "Fitxer de configuració erroni — no s’han trobat APNAME ni PASSWORD a %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_chs.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_chs.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_chs.h	2023-04-04 05:39:02.882217185 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_chs.h	2023-04-04 05:31:23.447652065 +0200
@@ -12475,18 +12475,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "提高或降低屏幕亮度。"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU 超频"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "超频或降频 Switch 的 GPU。"
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU 超频"
@@ -12574,6 +12563,8 @@
    "错误配置文件——在 %s 中找不到 APNAME 或 PASSWORD"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_cht.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_cht.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_cht.h	2023-04-04 05:39:02.886217276 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_cht.h	2023-04-04 05:34:45.436315456 +0200
@@ -14560,17 +14560,7 @@
    "調整螢幕亮度。"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU超頻"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Switch主機圖形處理器超頻設定。"
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU超頻"
@@ -14670,6 +14660,8 @@
    "錯誤的設定檔 - 遺失名稱或密碼 %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_cs.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_cs.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_cs.h	2023-04-04 05:39:02.890217368 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_cs.h	2023-04-04 05:31:23.463652435 +0200
@@ -14031,18 +14031,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Zvýšení Nebo Snížení Jasu Obrazovky."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Přetaktování GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Přetaktování nebo podtaktování přepínače grafického procesoru."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Přetaktování CPU"
@@ -14130,6 +14119,8 @@
    "Chybný konfigurační soubor - nepodařilo se najít APNAME nebo HESLO v %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_cy.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_cy.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_cy.h	2023-04-04 05:39:02.894217460 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_cy.h	2023-04-04 05:31:23.467652527 +0200
@@ -1723,13 +1723,11 @@
    MSG_LOCAL,
    "Lleol"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_da.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_da.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_da.h	2023-04-04 05:39:02.894217460 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_da.h	2023-04-04 05:31:23.467652527 +0200
@@ -1456,12 +1456,12 @@
    "Genstart RetroArch"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_de.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_de.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_de.h	2023-04-04 05:39:02.898217551 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_de.h	2023-04-04 05:31:23.467652527 +0200
@@ -13904,17 +13904,7 @@
    "Bildschirmhelligkeit erhöhen oder verringern."
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU Übertakten"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Über- oder Untertakten der Switch GPU."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU Übertakten"
@@ -14010,6 +14000,8 @@
    "Falsche Konfigurationsdatei - APNAME oder PASSWORD in %s nicht gefunden"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_el.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_el.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_el.h	2023-04-04 05:39:02.902217643 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_el.h	2023-04-04 05:31:23.467652527 +0200
@@ -5587,18 +5587,7 @@
    MENU_ENUM_LABEL_VALUE_BRIGHTNESS_CONTROL,
    "Φωτεινότητα Οθόνης"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Υπερχρονισμός Κάρτας Γραφικών"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Υπερχρονισμός ή υποχρονισμός της Κάρτας Γραφικών του Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Υπερχρονισμός Επεξεργαστή"
@@ -5646,6 +5635,8 @@
    "Επιλέξτε τη ζώνη ώρας σας για να προσαρμόσετε την ημερομηνία και την ώρα στην τοποθεσία σας."
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_en.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_en.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_en.h	2023-04-04 05:39:02.902217643 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_en.h	2023-04-04 05:31:23.467652527 +0200
@@ -2156,17 +2156,7 @@
    "Use the lowest frequency available to save power. Useful on battery powered devices, but performance will be significantly reduced."
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU Over-clock"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Over-clock or under-clock the Switch GPU."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU Over-clock"
@@ -2178,6 +2168,8 @@
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_SUBLABEL_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_eo.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_eo.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_eo.h	2023-04-04 05:39:02.902217643 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_eo.h	2023-04-04 05:31:23.467652527 +0200
@@ -1127,10 +1127,7 @@
    MENU_ENUM_LABEL_VALUE_INPUT_SMALL_KEYBOARD_ENABLE,
    "Small Keyboard Enable"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -1154,6 +1151,8 @@
    "Enable or disable remote command line access."
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_es.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_es.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_es.h	2023-04-04 05:39:02.906217735 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_es.h	2023-04-04 05:31:23.467652527 +0200
@@ -14227,18 +14227,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Aumenta o reduce el brillo de la pantalla."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Acelerar GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Aumenta o reduce la velocidad de la GPU de Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Acelerar CPU"
@@ -14330,6 +14319,8 @@
    "Archivo de configuración incorrecto: no se encontraron los valores APNAME o PASSWORD en %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_fa.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_fa.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_fa.h	2023-04-04 05:39:02.910217827 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_fa.h	2023-04-04 05:31:23.467652527 +0200
@@ -2660,9 +2660,7 @@
    "ریستارت رتروآرچ"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -2670,6 +2668,8 @@
    "بلوتوث"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_fi.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_fi.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_fi.h	2023-04-04 05:39:02.910217827 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_fi.h	2023-04-04 05:31:23.471652620 +0200
@@ -13171,18 +13171,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Lisää tai vähennä näytön kirkkautta."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Näytönohjaimen ylikellotus"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Ylikellota tai alikellota Switchin näytönohjain."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Suorittimen ylikellotus"
@@ -13266,6 +13255,8 @@
    "Väärä kokoonpanotiedosto- APNAME tai PASSWORD ei löydy kohteesta %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_fr.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_fr.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_fr.h	2023-04-04 05:39:02.914217918 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_fr.h	2023-04-04 05:31:23.471652620 +0200
@@ -14179,18 +14179,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Augmenter ou diminuer la luminosité de l'écran."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Overclocker le processeur graphique"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Overclocker ou underclocker le processeur graphique de la Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Overclocker le processeur"
@@ -14274,6 +14263,8 @@
    "Mauvais fichier de configuration - impossible de trouver l'APNAME ou le PASSWORD dans %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_gl.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_gl.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_gl.h	2023-04-04 05:39:02.918218010 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_gl.h	2023-04-04 05:31:23.471652620 +0200
@@ -2175,10 +2175,7 @@
    MENU_ENUM_LABEL_VALUE_INPUT_META_RESTART_KEY,
    "Reiniciar RetroArch"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -2206,6 +2203,8 @@
    "Desconectando da wifi \"%s\""
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_he.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_he.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_he.h	2023-04-04 05:39:02.918218010 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_he.h	2023-04-04 05:31:23.471652620 +0200
@@ -900,12 +900,12 @@
    "אתחול RetroArch"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_hr.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_hr.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_hr.h	2023-04-04 05:39:02.918218010 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_hr.h	2023-04-04 05:31:23.471652620 +0200
@@ -1104,12 +1104,12 @@
    "Ponovno pokreni RetroArch"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_hu.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_hu.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_hu.h	2023-04-04 05:39:02.922218102 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_hu.h	2023-04-04 05:31:23.471652620 +0200
@@ -13995,18 +13995,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "A kijelző fényerejének növelése vagy csökkentése."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU túlhajtás"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "A Switch GPU-jának túlhajtása vagy visszafogása."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU túlhajtás"
@@ -14094,6 +14083,8 @@
    "Rossz konfigurációs fájl - nincs APNAME vagy PASSWORD itt: %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_id.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_id.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_id.h	2023-04-04 05:39:02.922218102 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_id.h	2023-04-04 05:31:23.471652620 +0200
@@ -3887,18 +3887,7 @@
    MENU_ENUM_LABEL_VALUE_INPUT_KEYBOARD_GAMEPAD_MAPPING_TYPE,
    "Jenis Pemetaan Stik Kendali Papan Ketik"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Percepatan GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Mempercepat atau perlambat GPU permainan Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Percepatan CPU"
@@ -3914,6 +3903,8 @@
    "Memutus sambungan dari Wi-Fi '%s'"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_it.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_it.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_it.h	2023-04-04 05:39:02.926218193 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_it.h	2023-04-04 05:31:23.475652712 +0200
@@ -14067,18 +14067,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Aumenta o diminuisce la luminosità dello schermo."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Overclocca la GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Overclocca o underclocca la GPU del Nintendo Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Overclocca la CPU"
@@ -14174,6 +14163,8 @@
    "File di configurazione errato: impossibile trovare APNAME o PASSWORD in %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ja.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ja.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ja.h	2023-04-04 05:39:02.930218285 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ja.h	2023-04-04 05:31:23.475652712 +0200
@@ -11127,18 +11127,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "画面の明るさを増減します."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPUオーバークロック"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "SwitchのGPUをオーバークロック/アンダークロックします."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPUオーバークロック"
@@ -11230,6 +11219,8 @@
    "設定ファイルが不正です - APNAME または PASSWORD が %s に見つかりませんでした"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ko.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ko.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ko.h	2023-04-04 05:39:02.934218377 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ko.h	2023-04-04 05:31:23.475652712 +0200
@@ -14280,17 +14280,7 @@
    "화면의 밝기를 조절합니다."
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU 오버클럭"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "스위치 GPU를 오버클럭 또는 언더클럭합니다."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU 오버클럭"
@@ -14390,6 +14380,8 @@
    "잘못된 구성 파일 - %s에서 APNAME 또는 PASSWORD를 찾을 수 없습니다."
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_lbl.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_lbl.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_lbl.h	2023-04-04 05:39:02.942218560 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_lbl.h	2023-04-04 05:31:23.475652712 +0200
@@ -1,10 +1,4 @@
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_SWITCH_GPU_PROFILE,
-   "switch_gpu_profile"
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_SWITCH_CPU_PROFILE,
    "switch_cpu_profile"
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_mt.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_mt.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_mt.h	2023-04-04 05:39:02.942218560 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_mt.h	2023-04-04 05:31:23.475652712 +0200
@@ -423,13 +423,12 @@
 
 /* Environment Specific Settings */
 
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_nl.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_nl.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_nl.h	2023-04-04 05:39:02.942218560 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_nl.h	2023-04-04 05:31:23.475652712 +0200
@@ -6115,10 +6115,7 @@
    MENU_ENUM_LABEL_VALUE_INPUT_SMALL_KEYBOARD_ENABLE,
    "Small Keyboard Enable"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -6142,6 +6139,8 @@
    "Enable or disable remote command line access."
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_no.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_no.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_no.h	2023-04-04 05:39:02.942218560 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_no.h	2023-04-04 05:31:23.475652712 +0200
@@ -2467,13 +2467,12 @@
    MENU_ENUM_LABEL_VALUE_INPUT_META_RESTART_KEY,
    "Start RetroArch på nytt"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_oc.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_oc.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_oc.h	2023-04-04 05:39:02.942218560 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_oc.h	2023-04-04 05:31:23.475652712 +0200
@@ -667,13 +667,11 @@
    MENU_ENUM_LABEL_VALUE_INPUT_META_RESTART_KEY,
    "Reaviar RetroArch"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_pl.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_pl.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_pl.h	2023-04-04 05:39:02.946218652 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_pl.h	2023-04-04 05:31:23.479652804 +0200
@@ -12743,18 +12743,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Zwiększ lub zmniejsz jasność ekranu."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Podkręcanie GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Podkręcanie lub obniżanie GPU."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Podkręcanie CPU"
@@ -12850,6 +12839,8 @@
    "Niepoprawny plik konfiguracyjny - nie można znaleźć APNAME lub PASSWORD w %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_pt_br.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_pt_br.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_pt_br.h	2023-04-04 05:39:02.950218744 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_pt_br.h	2023-04-04 05:31:23.479652804 +0200
@@ -12979,18 +12979,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Aumenta ou diminui o brilho da tela."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Overclock da GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Faz um overclock ou underclock na CPU do Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Overclock da CPU"
@@ -13086,6 +13075,8 @@
    "Arquivo de configuração incorreto: valores APNAME ou PASSWORD não puderam ser encontrados em %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_pt_pt.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_pt_pt.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_pt_pt.h	2023-04-04 05:39:02.954218835 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_pt_pt.h	2023-04-04 05:31:23.479652804 +0200
@@ -5256,9 +5256,7 @@
    "Ativar teclado pequeno"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -5290,6 +5288,8 @@
    "Ativar/desativar acesso remoto pela linha de comandos."
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ro.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ro.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ro.h	2023-04-04 05:39:02.954218835 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ro.h	2023-04-04 05:31:23.479652804 +0200
@@ -424,12 +424,12 @@
 /* Environment Specific Settings */
 
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ru.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ru.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_ru.h	2023-04-04 05:39:02.954218835 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_ru.h	2023-04-04 05:31:23.479652804 +0200
@@ -14307,18 +14307,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Увеличить или уменьшить яркость экрана."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Разгон GPU"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Разгон или замедление графического процессора Switch."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Разгон СPU"
@@ -14406,6 +14395,8 @@
    "Неправильный файл конфигурации - отсутствует APNAME или PASSWORD в %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_si.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_si.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_si.h	2023-04-04 05:39:02.958218927 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_si.h	2023-04-04 05:31:23.479652804 +0200
@@ -516,12 +516,12 @@
 /* Environment Specific Settings */
 
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_sk.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_sk.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_sk.h	2023-04-04 05:39:02.962219019 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_sk.h	2023-04-04 05:31:23.479652804 +0200
@@ -5763,18 +5763,7 @@
    MSG_LOCAL,
    "Miestne"
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU pretaktovanie"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Pretaktovanie alebo podtaktovanie GPU Switcha."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU pretaktovanie"
@@ -5802,6 +5791,8 @@
    "[Žiadne SSID]"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_sr.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_sr.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_sr.h	2023-04-04 05:39:03.018220303 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_sr.h	2023-04-04 05:31:23.483652897 +0200
@@ -2424,12 +2424,12 @@
    "Restartuj RetroArch"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_sv.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_sv.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_sv.h	2023-04-04 05:39:03.018220303 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_sv.h	2023-04-04 05:31:23.483652897 +0200
@@ -4056,9 +4056,7 @@
    "Lokal"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -4070,6 +4068,8 @@
    "Tidszon"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_tr.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_tr.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_tr.h	2023-04-04 05:39:03.022220394 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_tr.h	2023-04-04 05:31:23.483652897 +0200
@@ -13423,18 +13423,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Ekran parlaklığını arttır veya azalt."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU Hız Aşırtma"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "GPU hızını arttır yada düşür."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU Hız Aşırtma"
@@ -13522,6 +13511,8 @@
    "Yanlış yapılandırma dosyası - %s içinde APNAME veya PASSWORD bulunamadı"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_uk.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_uk.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_uk.h	2023-04-04 05:39:03.026220486 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_uk.h	2023-04-04 05:31:23.483652897 +0200
@@ -8172,13 +8172,7 @@
    "Місцеві"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "Розгін GPU"
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "Розгін CPU"
@@ -8210,6 +8204,8 @@
    "Точка доступу Wi-Fi не запущена"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_us.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_us.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_us.h	2023-04-04 05:39:03.026220486 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_us.h	2023-04-04 05:31:23.483652897 +0200
@@ -15031,18 +15031,7 @@
    MENU_ENUM_SUBLABEL_BRIGHTNESS_CONTROL,
    "Increase or decrease the screen brightness."
    )
-
-#ifdef HAVE_LAKKA_SWITCH
-MSG_HASH(
-   MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-   "GPU Overclock"
-   )
-MSG_HASH(
-   MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE,
-   "Overclock or underclock the Switch GPU."
-   )
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE,
    "CPU Overclock"
@@ -15142,6 +15131,8 @@
    "Wrong configuration file - could not find APNAME or PASSWORD in %s"
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_SCALE,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_val.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_val.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_val.h	2023-04-04 05:39:03.030220578 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_val.h	2023-04-04 05:31:23.483652897 +0200
@@ -3380,12 +3380,12 @@
    "Reinicia RetroArch"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_vn.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_vn.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/intl/msg_hash_vn.h	2023-04-04 05:39:03.030220578 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/intl/msg_hash_vn.h	2023-04-04 05:31:23.483652897 +0200
@@ -2600,9 +2600,7 @@
    "Small Keyboard Enable"
    )
 
-#ifdef HAVE_LAKKA_SWITCH
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
 #endif
 #ifdef HAVE_LAKKA
 MSG_HASH(
@@ -2630,6 +2628,8 @@
    "Bật/tắt giao thức SSH."
    )
 #endif
+#ifdef HAVE_LAKKA_SWITCH
+#endif
 #ifdef GEKKO
 #endif
 #ifdef HAVE_ODROIDGO2
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_deferred_push.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_deferred_push.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_deferred_push.c	2023-04-04 05:39:03.034220670 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_deferred_push.c	2023-04-04 05:31:23.487652990 +0200
@@ -256,14 +256,10 @@
 GENERIC_DEFERRED_PUSH(deferred_push_lakka_list,                     DISPLAYLIST_LAKKA)
 #endif
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
 GENERIC_DEFERRED_PUSH(deferred_push_switch_cpu_profile,             DISPLAYLIST_SWITCH_CPU_PROFILE)
 #endif
 
-#ifdef HAVE_LAKKA_SWITCH
-GENERIC_DEFERRED_PUSH(deferred_push_switch_gpu_profile,             DISPLAYLIST_SWITCH_GPU_PROFILE)
-#endif
-
 #if defined(HAVE_LAKKA)
 GENERIC_DEFERRED_PUSH(deferred_push_cpu_perfpower,                  DISPLAYLIST_CPU_PERFPOWER_LIST)
 GENERIC_DEFERRED_PUSH(deferred_push_cpu_policy,                     DISPLAYLIST_CPU_POLICY_LIST)
@@ -777,10 +773,7 @@
       {MENU_ENUM_LABEL_DEFERRED_AUDIO_RESAMPLER_SETTINGS_LIST, deferred_push_audio_resampler_settings_list},
       {MENU_ENUM_LABEL_DEFERRED_AUDIO_MIXER_SETTINGS_LIST, deferred_push_audio_mixer_settings_list},
       {MENU_ENUM_LABEL_DEFERRED_LATENCY_SETTINGS_LIST, deferred_push_latency_settings_list},
-#ifdef HAVE_LAKKA_SWITCH
-      {MENU_ENUM_LABEL_SWITCH_GPU_PROFILE, deferred_push_switch_gpu_profile},
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if  defined(HAVE_LIBNX)
       {MENU_ENUM_LABEL_SWITCH_CPU_PROFILE, deferred_push_switch_cpu_profile},
 #endif
 #if defined(HAVE_LAKKA)
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_get_value.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_get_value.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_get_value.c	2023-04-04 05:39:03.034220670 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_get_value.c	2023-04-04 05:31:23.487652990 +0200
@@ -599,7 +599,6 @@
       strlcpy(s2, alt, len2);
 }
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
 static void menu_action_setting_disp_cpu_gov_mode(
       file_list_t* list,
@@ -764,7 +763,6 @@
    strlcpy(s, d->scaling_governor, len);
 }
 #endif
-#endif
 
 static void menu_action_setting_disp_set_label_core_lock(
       file_list_t* list,
@@ -2026,7 +2024,6 @@
             BIND_ACTION_GET_VALUE(cbs,
                   menu_action_setting_disp_set_label_core_option_override_info);
             break;
-         #ifndef HAVE_LAKKA_SWITCH
          #ifdef HAVE_LAKKA
          case MENU_ENUM_LABEL_CPU_PERF_MODE:
             BIND_ACTION_GET_VALUE(cbs,
@@ -2053,7 +2050,6 @@
             BIND_ACTION_GET_VALUE(cbs, menu_action_cpu_governor_label);
             break;
          #endif
-         #endif
          default:
             return -1;
       }
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_left.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_left.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_left.c	2023-04-04 05:39:03.034220670 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_left.c	2023-04-04 05:31:23.487652990 +0200
@@ -686,7 +686,6 @@
    return 0;
 }
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
 static int cpu_policy_mode_change(unsigned type, const char *label,
       bool wraparound)
@@ -801,7 +800,6 @@
    return 0;
 }
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
 
 static int core_setting_left(unsigned type, const char *label,
       bool wraparound)
@@ -1129,7 +1127,6 @@
             case MENU_ENUM_LABEL_MANUAL_CONTENT_SCAN_CORE_NAME:
                BIND_ACTION_LEFT(cbs, manual_content_scan_core_name_left);
                break;
-            #ifndef HAVE_LAKKA_SWITCH
             #ifdef HAVE_LAKKA
             case MENU_ENUM_LABEL_CPU_PERF_MODE:
                BIND_ACTION_LEFT(cbs, cpu_policy_mode_change);
@@ -1148,7 +1145,6 @@
                BIND_ACTION_LEFT(cbs, cpu_policy_freq_managed_gov);
                break;
             #endif
-            #endif
             default:
                return -1;
          }
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_ok.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_ok.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_ok.c	2023-04-04 05:39:03.034220670 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_ok.c	2023-04-04 05:31:23.487652990 +0200
@@ -100,7 +100,7 @@
 #include "../../play_feature_delivery/play_feature_delivery.h"
 #endif
 
-#if defined(HAVE_LAKKA) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
 #include "../../switch_performance_profiles.h"
 #endif
 
@@ -3874,19 +3874,11 @@
    return 0;
 }
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
 static int action_ok_set_switch_cpu_profile(const char *path,
       const char *label, unsigned type, size_t idx, size_t entry_idx)
 {
    char command[PATH_MAX_LENGTH] = {0};
-#ifdef HAVE_LAKKA_SWITCH
-   char* profile_name            = SWITCH_CPU_PROFILES[entry_idx];
-
-   snprintf(command, sizeof(command), "cpu-profile set '%s'", profile_name);
-
-   system(command);
-   snprintf(command, sizeof(command), "Current profile set to %s", profile_name);
-#else
    unsigned profile_clock          = SWITCH_CPU_SPEEDS_VALUES[entry_idx];
    settings_t *settings            = config_get_ptr();
 
@@ -3903,7 +3895,6 @@
    }
    snprintf(command, sizeof(command),
          "Current Clock set to %i", profile_clock);
-#endif
 
    runloop_msg_queue_push(command, 1, 90, true, NULL,
          MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
@@ -3912,33 +3903,6 @@
 }
 #endif
 
-#ifdef HAVE_LAKKA_SWITCH
-
-static int action_ok_set_switch_gpu_profile(const char *path,
-      const char *label, unsigned type, size_t idx, size_t entry_idx)
-{
-   char command[PATH_MAX_LENGTH];
-   char            *profile_name  = SWITCH_GPU_PROFILES[entry_idx];
-
-   command[0]                     = '\0';
-
-   snprintf(command, sizeof(command),
-         "gpu-profile set '%s'",
-         profile_name);
-
-   system(command);
-
-   snprintf(command, sizeof(command),
-         "Current profile set to %s",
-         profile_name);
-
-   runloop_msg_queue_push(command, 1, 90, true, NULL,
-         MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
-
-   return -1;
-}
-
-#endif
 
 static int action_ok_load_core_deferred(const char *path,
       const char *label, unsigned type, size_t idx, size_t entry_idx)
@@ -8266,10 +8230,8 @@
          {MENU_ENUM_LABEL_NETPLAY_REFRESH_LAN,                 action_ok_push_netplay_refresh_lan},
 #endif
 #endif
-#ifdef HAVE_LAKKA_SWITCH
-         {MENU_ENUM_LABEL_SWITCH_GPU_PROFILE,                  action_ok_push_default},
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+
+#if defined(HAVE_LIBNX)
          {MENU_ENUM_LABEL_SWITCH_CPU_PROFILE,                  action_ok_push_default},
 #endif
          {MENU_ENUM_LABEL_MENU_WALLPAPER,                      action_ok_menu_wallpaper},
@@ -8790,12 +8752,7 @@
          case FILE_TYPE_PLAYLIST_ENTRY:
             BIND_ACTION_OK(cbs, action_ok_playlist_entry_collection);
             break;
-#ifdef HAVE_LAKKA_SWITCH
-         case MENU_SET_SWITCH_GPU_PROFILE:
-            BIND_ACTION_OK(cbs, action_ok_set_switch_gpu_profile);
-            break;
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
          case MENU_SET_SWITCH_CPU_PROFILE:
             BIND_ACTION_OK(cbs, action_ok_set_switch_cpu_profile);
             break;
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_right.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_right.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_right.c	2023-04-04 05:39:03.038220761 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_right.c	2023-04-04 05:31:23.487652990 +0200
@@ -795,7 +795,6 @@
    return 0;
 }
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
 static int cpu_policy_mode_change(unsigned type, const char *label,
       bool wraparound)
@@ -913,7 +912,7 @@
    return 0;
 }
 #endif
-#endif
+
 int core_setting_right(unsigned type, const char *label,
       bool wraparound)
 {
@@ -1257,7 +1256,6 @@
             case MENU_ENUM_LABEL_MANUAL_CONTENT_SCAN_CORE_NAME:
                BIND_ACTION_RIGHT(cbs, manual_content_scan_core_name_right);
                break;
-            #ifndef HAVE_LAKKA_SWITCH
             #ifdef HAVE_LAKKA
             case MENU_ENUM_LABEL_CPU_PERF_MODE:
                BIND_ACTION_RIGHT(cbs, cpu_policy_mode_change);
@@ -1276,7 +1274,6 @@
                BIND_ACTION_RIGHT(cbs, cpu_policy_freq_managed_gov);
                break;
             #endif
-            #endif
             default:
                return -1;
          }
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_sublabel.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_sublabel.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_sublabel.c	2023-04-04 05:39:03.038220761 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_sublabel.c	2023-04-04 05:31:23.487652990 +0200
@@ -1155,14 +1155,10 @@
 #endif
 DEFAULT_SUBLABEL_MACRO(action_bind_sublabel_discord_allow,                         MENU_ENUM_SUBLABEL_DISCORD_ALLOW)
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
 DEFAULT_SUBLABEL_MACRO(action_bind_sublabel_switch_cpu_profile,             MENU_ENUM_SUBLABEL_SWITCH_CPU_PROFILE)
 #endif
 
-#ifdef HAVE_LAKKA_SWITCH
-DEFAULT_SUBLABEL_MACRO(action_bind_sublabel_switch_gpu_profile,             MENU_ENUM_SUBLABEL_SWITCH_GPU_PROFILE)
-#endif
-
 #ifdef __linux__
 DEFAULT_SUBLABEL_MACRO(action_bind_sublabel_gamemode_enable,                MENU_ENUM_SUBLABEL_GAMEMODE_ENABLE_LINUX)
 #else
@@ -1305,7 +1301,6 @@
 }
 #endif
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
 static int action_bind_sublabel_cpu_policy_entry_list(
       file_list_t *list,
@@ -1338,7 +1333,7 @@
    return 0;
 }
 #endif
-#endif
+
 #ifdef HAVE_CHEEVOS
 static int action_bind_sublabel_cheevos_entry(
       file_list_t *list,
@@ -4868,7 +4863,6 @@
          case MENU_ENUM_LABEL_TIMEZONE:
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_timezone);
             break;
-#ifndef HAVE_LAKKA_SWITCH
          case MENU_ENUM_LABEL_CPU_POLICY_ENTRY:
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_cpu_policy_entry_list);
             break;
@@ -4876,7 +4870,6 @@
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_cpu_perf_mode);
             break;
 #endif
-#endif
          case MENU_ENUM_LABEL_USER_LANGUAGE:
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_user_language);
             break;
@@ -4960,16 +4953,11 @@
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_show_wimp);
             break;
 #endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
          case MENU_ENUM_LABEL_SWITCH_CPU_PROFILE:
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_switch_cpu_profile);
             break;
 #endif
-#ifdef HAVE_LAKKA_SWITCH
-         case MENU_ENUM_LABEL_SWITCH_GPU_PROFILE:
-            BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_switch_gpu_profile);
-            break;
-#endif
          case MENU_ENUM_LABEL_GAMEMODE_ENABLE:
             BIND_ACTION_SUBLABEL(cbs, action_bind_sublabel_gamemode_enable);
             break;
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_title.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_title.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/cbs/menu_cbs_title.c	2023-04-04 05:39:03.042220853 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/cbs/menu_cbs_title.c	2023-04-04 05:31:23.487652990 +0200
@@ -765,14 +765,10 @@
 DEFAULT_TITLE_COPY_MACRO(action_get_title_video_shader_preset_remove,MENU_ENUM_LABEL_VALUE_VIDEO_SHADER_PRESET_REMOVE)
 DEFAULT_TITLE_COPY_MACRO(action_get_title_video_shader_preset_save_list,MENU_ENUM_LABEL_VALUE_VIDEO_SHADER_PRESET_SAVE)
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
 DEFAULT_TITLE_MACRO(action_get_title_switch_cpu_profile,          MENU_ENUM_LABEL_VALUE_SWITCH_CPU_PROFILE)
 #endif
 
-#ifdef HAVE_LAKKA_SWITCH
-DEFAULT_TITLE_MACRO(action_get_title_switch_gpu_profile,          MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE)
-#endif
-
 DEFAULT_TITLE_SEARCH_FILTER_MACRO(action_get_title_deferred_history_list,   MENU_ENUM_LABEL_VALUE_LOAD_CONTENT_HISTORY)
 DEFAULT_TITLE_SEARCH_FILTER_MACRO(action_get_title_deferred_favorites_list, MENU_ENUM_LABEL_VALUE_GOTO_FAVORITES)
 DEFAULT_TITLE_SEARCH_FILTER_MACRO(action_get_title_deferred_images_list,    MENU_ENUM_LABEL_VALUE_GOTO_IMAGES)
@@ -1285,14 +1281,10 @@
          action_get_title_core_directory},
       {MENU_ENUM_LABEL_LIBRETRO_INFO_PATH,
          action_get_title_core_info_directory},
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
       {MENU_ENUM_LABEL_SWITCH_CPU_PROFILE,
          action_get_title_switch_cpu_profile},
 #endif
-#ifdef HAVE_LAKKA_SWITCH
-      {MENU_ENUM_LABEL_SWITCH_GPU_PROFILE,
-         action_get_title_switch_gpu_profile},
-#endif
       {MENU_ENUM_LABEL_DEFERRED_MANUAL_CONTENT_SCAN_LIST,
          action_get_title_manual_content_scan_list},
       {MENU_ENUM_LABEL_MANUAL_CONTENT_SCAN_DIR,
@@ -1716,16 +1708,11 @@
          case MENU_ENUM_LABEL_LIBRETRO_INFO_PATH:
             BIND_ACTION_GET_TITLE(cbs, action_get_title_core_info_directory);
             break;
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
          case MENU_ENUM_LABEL_SWITCH_CPU_PROFILE:
             BIND_ACTION_GET_TITLE(cbs, action_get_title_switch_cpu_profile);
             break;
 #endif
-#ifdef HAVE_LAKKA_SWITCH
-         case MENU_ENUM_LABEL_SWITCH_GPU_PROFILE:
-            BIND_ACTION_GET_TITLE(cbs, action_get_title_switch_gpu_profile);
-            break;
-#endif
          case MENU_ENUM_LABEL_MANUAL_CONTENT_SCAN_LIST:
             BIND_ACTION_GET_TITLE(cbs, action_get_title_manual_content_scan_list);
             break;
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/drivers/ozone.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/drivers/ozone.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/drivers/ozone.c	2023-04-04 05:39:03.042220853 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/drivers/ozone.c	2023-04-04 05:31:23.487652990 +0200
@@ -2053,10 +2053,7 @@
       case MENU_ENUM_LABEL_USER_INTERFACE_SETTINGS:
       case MENU_ENUM_LABEL_SETTINGS_SHOW_USER_INTERFACE:
             return ozone->icons_textures[OZONE_ENTRIES_ICONS_TEXTURE_UI];
-#ifdef HAVE_LAKKA_SWITCH
-      case MENU_ENUM_LABEL_SWITCH_GPU_PROFILE:
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
       case MENU_ENUM_LABEL_SWITCH_CPU_PROFILE:
             return ozone->icons_textures[OZONE_ENTRIES_ICONS_TEXTURE_POWER];
 #endif
@@ -9364,21 +9361,13 @@
                      false);
             }
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
             MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(
                   info->list,
                   MENU_ENUM_LABEL_SWITCH_CPU_PROFILE,
                   PARSE_ACTION,
                   false);
 #endif
-
-#ifdef HAVE_LAKKA_SWITCH
-            MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(
-                  info->list,
-                  MENU_ENUM_LABEL_SWITCH_GPU_PROFILE,
-                  PARSE_ACTION,
-                  false);
-#endif
 
             if (settings->bools.menu_show_configurations && !settings->bools.kiosk_mode_enable)
             {
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/drivers/xmb.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/drivers/xmb.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/drivers/xmb.c	2023-04-04 05:39:03.046220945 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/drivers/xmb.c	2023-04-04 05:31:23.491653081 +0200
@@ -3185,10 +3185,7 @@
       case MENU_ENUM_LABEL_USER_INTERFACE_SETTINGS:
       case MENU_ENUM_LABEL_SETTINGS_SHOW_USER_INTERFACE:
          return xmb->textures.list[XMB_TEXTURE_UI];
-#ifdef HAVE_LAKKA_SWITCH
-      case MENU_ENUM_LABEL_SWITCH_GPU_PROFILE:
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
       case MENU_ENUM_LABEL_SWITCH_CPU_PROFILE:
          return xmb->textures.list[XMB_TEXTURE_POWER];
 #endif
@@ -8092,22 +8089,13 @@
                      false);
             }
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
             MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(
                   info->list,
                   MENU_ENUM_LABEL_SWITCH_CPU_PROFILE,
                   PARSE_ACTION,
                   false);
 #endif
-
-#ifdef HAVE_LAKKA_SWITCH
-            MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(
-                  info->list,
-                  MENU_ENUM_LABEL_SWITCH_GPU_PROFILE,
-                  PARSE_ACTION,
-                  false);
-#endif
-
             if (menu_show_configurations && !kiosk_mode_enable)
             {
                MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_displaylist.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_displaylist.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_displaylist.c	2023-04-04 05:39:03.050221036 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_displaylist.c	2023-04-04 05:31:23.491653081 +0200
@@ -50,7 +50,7 @@
 #include "../core_updater_list.h"
 #endif
 
-#ifdef HAVE_LAKKA_SWITCH
+#ifdef HAVE_LAKKA
 #include "../../lakka.h"
 #endif
 
@@ -58,7 +58,7 @@
 #include <switch.h>
 #endif
 
-#if defined(HAVE_LAKKA) || defined(HAVE_LIBNX)
+#if /*defined(HAVE_LAKKA) ||*/ defined(HAVE_LIBNX)
 #include "../../switch_performance_profiles.h"
 #endif
 
@@ -11690,7 +11690,6 @@
          /* No-op */
          break;
 #endif
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       case DISPLAYLIST_CPU_POLICY_LIST:
          menu_entries_ctl(MENU_ENTRIES_CTL_CLEAR, info->list);
@@ -11792,30 +11791,17 @@
          break;
       }
 #endif
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
       case DISPLAYLIST_SWITCH_CPU_PROFILE:
       {
          unsigned i;
          char text[PATH_MAX_LENGTH];
-#ifdef HAVE_LAKKA_SWITCH
-         char current_profile[PATH_MAX_LENGTH];
-         FILE               *profile = NULL;
-#endif
          const size_t profiles_count = sizeof(SWITCH_CPU_PROFILES)/sizeof(SWITCH_CPU_PROFILES[1]);
 
          runloop_msg_queue_push("Warning : extended overclocking can damage the Switch", 1, 90, true, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
 
          menu_entries_ctl(MENU_ENTRIES_CTL_CLEAR, info->list);
 
-#ifdef HAVE_LAKKA_SWITCH
-         profile = popen("cpu-profile get", "r");
-         fgets(current_profile, PATH_MAX_LENGTH, profile);
-         pclose(profile);
-
-         snprintf(text, sizeof(text),
-               "Current profile: %s", current_profile);
-#else
          u32 currentClock = 0;
          if (hosversionBefore(8, 0, 0))
             pcvGetClockRate(PcvModule_CpuBus, &currentClock);
@@ -11827,7 +11813,6 @@
             clkrstCloseSession(&session);
          }
          snprintf(text, sizeof(text), "Current Clock : %i", currentClock);
-#endif
          if (menu_entries_append(info->list,
             text,
             "",
@@ -11857,49 +11842,7 @@
                             | MD_FLAG_NEED_CLEAR;
          break;
       }
-#if defined(HAVE_LAKKA_SWITCH)
-      case DISPLAYLIST_SWITCH_GPU_PROFILE:
-      {
-         unsigned i;
-         char text[PATH_MAX_LENGTH];
-         char current_profile[PATH_MAX_LENGTH];
-         FILE               *profile = NULL;
-         const size_t profiles_count = sizeof(SWITCH_GPU_PROFILES)/sizeof(SWITCH_GPU_PROFILES[1]);
-
-         runloop_msg_queue_push("Warning : extended overclocking can damage the Switch", 1, 90, true, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
-
-         profile = popen("gpu-profile get", "r");
-         fgets(current_profile, PATH_MAX_LENGTH, profile);
-         pclose(profile);
-
-         menu_entries_ctl(MENU_ENTRIES_CTL_CLEAR, info->list);
-
-         /* TODO/FIXME - Localize */
-         snprintf(text, sizeof(text),
-               "Current profile : %s", current_profile);
-
-         if (menu_entries_append(info->list, text, "", 0, MENU_INFO_MESSAGE, 0, 0, NULL))
-            count++;
-
-         for (i = 0; i < profiles_count; i++)
-         {
-            char title[PATH_MAX_LENGTH];
-            char* profile               = SWITCH_GPU_PROFILES[i];
-            char* speed                 = SWITCH_GPU_SPEEDS[i];
-
-            snprintf(title, sizeof(title), "%s (%s)", profile, speed);
-
-            if (menu_entries_append(info->list, title, "", 0, MENU_SET_SWITCH_GPU_PROFILE, 0, i, NULL))
-               count++;
-         }
-
-         info->flags       |= MD_FLAG_NEED_REFRESH
-                            | MD_FLAG_NEED_PUSH
-                            | MD_FLAG_NEED_CLEAR;
-         break;
-      }
-#endif /* HAVE_LAKKA_SWITCH */
-#endif /* HAVE_LAKKA_SWITCH || HAVE_LIBNX */
+#endif /* HAVE_LIBNX */
       case DISPLAYLIST_MUSIC_LIST:
          {
             menu_entries_ctl(MENU_ENTRIES_CTL_CLEAR, info->list);
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_displaylist.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_displaylist.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_displaylist.h	2023-04-04 05:39:03.054221128 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_displaylist.h	2023-04-04 05:31:23.491653081 +0200
@@ -262,10 +262,7 @@
    DISPLAYLIST_CORE_CONTENT_DIRS,
    DISPLAYLIST_CORE_CONTENT_DIRS_SUBDIR,
    DISPLAYLIST_CORE_SYSTEM_FILES,
-#ifdef HAVE_LAKKA_SWITCH
-   DISPLAYLIST_SWITCH_GPU_PROFILE,
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
    DISPLAYLIST_SWITCH_CPU_PROFILE,
 #endif
    DISPLAYLIST_MANUAL_CONTENT_SCAN_LIST,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_driver.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_driver.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_driver.c	2023-04-04 05:39:03.054221128 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_driver.c	2023-04-04 05:31:23.491653081 +0200
@@ -74,9 +74,6 @@
 
 #ifdef HAVE_LIBNX
 #include <switch.h>
-#endif
-
-#if defined(HAVE_LAKKA) || defined(HAVE_LIBNX)
 #include "../switch_performance_profiles.h"
 #endif
 
@@ -6761,11 +6758,9 @@
 
    if (on) 
    {
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_MENU);
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
 #ifdef HAVE_OVERLAY
       /* If an overlay was displayed before the toggle
        * and overlays are disabled in menu, need to
@@ -6784,11 +6779,9 @@
    }
    else
    {
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_CORE);
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
 #ifdef HAVE_OVERLAY
       /* Inhibits pointer 'select' and 'cancel' actions
        * (until the next time 'select'/'cancel' are released) */
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_driver.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_driver.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_driver.h	2023-04-04 05:39:03.054221128 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_driver.h	2023-04-04 05:31:23.491653081 +0200
@@ -245,10 +245,7 @@
 
    MENU_SET_SCREEN_BRIGHTNESS,
 
-#ifdef HAVE_LAKKA_SWITCH
-   MENU_SET_SWITCH_GPU_PROFILE,
-#endif
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#if defined(HAVE_LIBNX)
    MENU_SET_SWITCH_CPU_PROFILE,
 #endif
 
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_setting.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_setting.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/menu/menu_setting.c	2023-04-04 05:39:03.058221220 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/menu/menu_setting.c	2023-04-04 05:31:23.495653174 +0200
@@ -8096,7 +8096,8 @@
             }
          }
          break;
-      case MENU_ENUM_LABEL_INPUT_POLL_TYPE_BEHAVIOR:
+
+     case MENU_ENUM_LABEL_INPUT_POLL_TYPE_BEHAVIOR:
          core_set_poll_type(*setting->value.target.integer);
          break;
       case MENU_ENUM_LABEL_VIDEO_SCALE_INTEGER:
@@ -10001,7 +10002,7 @@
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_QUIT);
 #endif
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
+#ifdef HAVE_LIBNX
         CONFIG_ACTION(
               list, list_info,
               MENU_ENUM_LABEL_SWITCH_CPU_PROFILE,
@@ -10012,15 +10013,6 @@
 #endif
 
 #if defined(HAVE_LAKKA)
-#ifdef HAVE_LAKKA_SWITCH
-        CONFIG_ACTION(
-               list, list_info,
-               MENU_ENUM_LABEL_SWITCH_GPU_PROFILE,
-               MENU_ENUM_LABEL_VALUE_SWITCH_GPU_PROFILE,
-               &group_info,
-               &subgroup_info,
-               parent_group);
-#endif
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_REBOOT,
@@ -18554,7 +18546,6 @@
 #endif
 
 #ifdef HAVE_LAKKA
-#ifndef HAVE_LAKKA_SWITCH
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_CPU_PERFPOWER,
@@ -18563,8 +18554,6 @@
                &subgroup_info,
                parent_group);
 #endif
-#endif
-
          if (frontend_driver_has_gamemode())
             CONFIG_BOOL(
                   list, list_info,
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/retroarch.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/retroarch.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/retroarch.c	2023-04-04 05:39:03.066221403 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/retroarch.c	2023-04-04 05:31:23.495653174 +0200
@@ -90,9 +90,6 @@
 
 #ifdef HAVE_LIBNX
 #include <switch.h>
-#endif
-
-#if defined(HAVE_LAKKA) || defined(HAVE_LIBNX)
 #include "switch_performance_profiles.h"
 #endif
 
@@ -960,11 +957,9 @@
    if (flags & DRIVER_MIDI_MASK)
       midi_driver_init(settings);
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
    cpu_scaling_driver_init();
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
 }
 
 void driver_uninit(int flags)
@@ -1058,11 +1053,9 @@
    if (flags & DRIVER_MIDI_MASK)
       midi_driver_free();
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
    cpu_scaling_driver_free();
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
 }
 
 static void retroarch_deinit_drivers(struct retro_callbacks *cbs)
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/runloop.c retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/runloop.c
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/runloop.c	2023-04-04 05:39:03.066221403 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/runloop.c	2023-04-04 05:31:23.495653174 +0200
@@ -97,9 +97,6 @@
 
 #ifdef HAVE_LIBNX
 #include <switch.h>
-#endif
-
-#if defined(HAVE_LAKKA) || defined(HAVE_LIBNX)
 #include "switch_performance_profiles.h"
 #endif
 
@@ -4599,19 +4596,15 @@
       command_event(CMD_EVENT_PRESENCE_UPDATE, &userdata);
 #endif
 
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_MENU);
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
    }
    else
    {
-#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_CORE);
 #endif
-#endif /* #ifndef HAVE_LAKKA_SWITCH */
    }
 
 #if defined(HAVE_TRANSLATE) && defined(HAVE_GFX_WIDGETS)
diff -Naur retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/switch_performance_profiles.h retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/switch_performance_profiles.h
--- retroarch-6616b807edea5b56683e9fbca8b003b436c22b13/switch_performance_profiles.h	2023-04-04 05:39:03.066221403 +0200
+++ retroarch-6616b807edea5b56683e9fbca8b003b436c22b13-2/switch_performance_profiles.h	2023-04-04 05:31:23.495653174 +0200
@@ -17,52 +17,9 @@
 #ifndef __SWITCH_PERFORMANCE_PROFILES_H
 #define __SWITCH_PERFORMANCE_PROFILES_H
 
-#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
-#ifdef HAVE_LAKKA_SWITCH
-static char *SWITCH_GPU_PROFILES[] = {
-    "Docked Stock +2",
-    "Docked Stock +1",
-    "Docked Stock Mode",
-    "Handheld Boost +3",
-    "Handheld Boost +2",
-    "Handheld Boost +1",
-    "Handheld Boost Mode",
-    "Handheld Stock +1",
-    "Handheld Stock Mode",
-    "Powersaving +2",
-    "Powersaving +1",
-    "Powersaving Mode",
-};
-
-static char *SWITCH_GPU_SPEEDS[] = {
-    "921 Mhz",
-    "844 Mhz",
-    "768 Mhz",
-    "691 Mhz",
-    "614 Mhz",
-    "537 Mhz",
-    "460 Mhz",
-    "384 Mhz",
-    "307 Mhz",
-    "230 Mhz",
-    "153 Mhz",
-    "76 Mhz"
-};
-#endif
+#if defined(HAVE_LIBNX)
 
 static char *SWITCH_CPU_PROFILES[] = {
-#ifndef HAVE_LIBNX
-    "Max Overdrive +3",
-    "Max Overdrive +2",
-    "Max Overdrive +1",
-    "Maximum Performance",
-    "High Performance",
-    "Boost Performance",
-    "Stock Performance",
-    "Powersaving Mode 1",
-    "Powersaving Mode 2",
-    "Powersaving Mode 3",
-#else
     "Maximum Performance",
     "High Performance",
     "Boost Performance",
@@ -70,25 +27,12 @@
     "Powersaving Mode 1",
     "Powersaving Mode 2",
     "Powersaving Mode 3",
-#endif
 };
 
 #define SWITCH_DEFAULT_CPU_PROFILE 3 /* Stock Performance */
 #define LIBNX_MAX_CPU_PROFILE 0 /* Max Performance */
 
 static char *SWITCH_CPU_SPEEDS[] = {
-#ifndef HAVE_LIBNX
-    " **UNSTABLE**  2091 MHz",
-    " **UNSTABLE**  1989 MHz",
-    " **UNSTABLE**  1887 MHz",
-    "1785 MHz",
-    "1581 MHz",
-    "1224 MHz",
-    "1020 MHz",
-    "918 MHz",
-    "816 MHz",
-    "714 MHz"
-#else
     "1785 MHz",
     "1581 MHz",
     "1224 MHz",
@@ -96,11 +40,8 @@
     "918 MHz",
     "816 MHz",
     "714 MHz"
-#endif
 };
 
-#ifdef HAVE_LIBNX
-// not used in switch-lakka
 static unsigned SWITCH_CPU_SPEEDS_VALUES[] = {
     1785000000,
     1581000000,
@@ -110,7 +51,6 @@
     816000000,
     714000000
 };
-#endif
 
 #endif
 
