diff -Naur retroarch-8618e55115aafc94f01181b16d90c7d46303df89/input/common/linux_common.c retroarch-8618e55115aafc94f01181b16d90c7d46303df89-2/input/common/linux_common.c
--- retroarch-8618e55115aafc94f01181b16d90c7d46303df89/input/common/linux_common.c	2025-12-02 03:15:18.368335043 +0100
+++ retroarch-8618e55115aafc94f01181b16d90c7d46303df89-2/input/common/linux_common.c	2025-12-02 03:17:00.981342840 +0100
@@ -320,7 +320,7 @@
    if (!in_illuminance_input)
    {
       RARCH_ERR("Failed to open \"%s\".\n", sensor->path);
-      return 0.0;
+      return -1.0;
    }
 
    /* Read the illuminance value from the file. If that fails... */
