diff -Naur upower-v1.90.0/src/linux/up-device-supply.c upower-v1.90.0-2/src/linux/up-device-supply.c
--- upower-v1.90.0/src/linux/up-device-supply.c	2024-02-13 04:47:45.985364890 +0100
+++ upower-v1.90.0-2/src/linux/up-device-supply.c	2024-02-18 21:30:18.926290971 +0100
@@ -517,9 +517,11 @@
 		     "power-supply", is_power_supply,
 		     NULL);
 
-	/* Handled by separate battery class */
-	if (is_power_supply)
-		g_assert (type == UP_DEVICE_KIND_LINE_POWER);
+	/* Not Handled by separate battery class */
+	if (is_power_supply && type == UP_DEVICE_KIND_LINE_POWER)
+		g_object_set (device, "poll-timeout", UP_DAEMON_UNKNOWN_TIMEOUT, NULL);
+        else if (is_power_supply)
+		g_assert (type != UP_DEVICE_KIND_BATTERY);
 
 	if (type != UP_DEVICE_KIND_LINE_POWER)
 		g_object_set (device, "poll-timeout", UP_DAEMON_SHORT_TIMEOUT, NULL);
