diff -Naur retroarch-9b282aa742b6c3d2f2925ae5a12e2cd7c6b6ad38/gfx/video_thread_wrapper.c retroarch-9b282aa742b6c3d2f2925ae5a12e2cd7c6b6ad38-2/gfx/video_thread_wrapper.c
--- retroarch-9b282aa742b6c3d2f2925ae5a12e2cd7c6b6ad38/gfx/video_thread_wrapper.c	2022-04-29 00:03:14.705913901 +0200
+++ retroarch-9b282aa742b6c3d2f2925ae5a12e2cd7c6b6ad38-2/gfx/video_thread_wrapper.c	2022-04-29 00:03:42.126313394 +0200
@@ -147,7 +147,7 @@
          {
             thr->driver_data = thr->driver->init(&thr->info,
                   thr->input, thr->input_data);
-            if (thr->driver->viewport_info)
+            if (thr->driver->viewport_info && thr->driver_data)
                thr->driver->viewport_info(thr->driver_data, &thr->vp);
          }
          else
@@ -471,7 +471,7 @@
          if (thr->driver && thr->driver->has_windowed)
             has_windowed = ret && thr->driver->has_windowed(thr->driver_data);
 
-         if (thr->driver && thr->driver->viewport_info)
+         if (thr->driver && thr->driver->viewport_info && thr->driver_data)
             thr->driver->viewport_info(thr->driver_data, &vp);
 
          slock_lock(thr->lock);
