diff --git a/Makefile b/Makefile
index 757c3bc..c0e21dd 100755
--- a/Makefile
+++ b/Makefile
@@ -61,10 +61,8 @@ else ifeq ($(platform), lakka-switch)
    DEFINES += -mcpu=cortex-a57+crypto+crc+fp+simd -mabi=lp64 -Wno-psabi -mtune=cortex-a57 \
 	 -march=armv8-a+crypto+crc+fp+simd -fomit-frame-pointer -Wall -pipe -fPIC -pthread \
 	 -D__LAKKA_SWITCH__
-   INCLUDES += -Ithird_party/opus/include -Ithird_party/ffmpeg/include \
-	 -I$(TOOLCHAIN)/aarch64-libreelec-linux-gnueabi/sysroot/usr/include -I$(TOOLCHAIN)/include
-   LIBS += -Lthird_party/opus/lib -Lthird_party/ffmpeg/lib \
-	 -lpthread -lGL -luuid -lvdpau -lX11
+   INCLUDES += -I$(TOOLCHAIN)/aarch64-libreelec-linux-gnueabi/sysroot/usr/include -I$(TOOLCHAIN)/include
+   LIBS += -lpthread -lGL -luuid -lvdpau -lX11
    SHARED := -shared -Wl,--version-script=link.T -Wl,--no-undefined
 else
    CC = gcc
